CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+cmake_minimum_required(VERSION 3.23)
+project(
+    cppfront
+    LANGUAGES CXX
+    VERSION 0.1.0
+)
+
+option(
+    CPPFRONT_INSTALL_RULES "Enable install rules for cppfront" "${PROJECT_IS_TOP_LEVEL}"
+)
+
+add_executable(cppfront source/cppfront.cpp)
+add_executable(cppfront::cppfront ALIAS cppfront)
+target_compile_features(cppfront PRIVATE cxx_std_20)
+target_sources(
+    cppfront
+    PRIVATE
+    FILE_SET HEADERS
+    BASE_DIRS source
+    FILES source/common.h source/lex.h source/load.h source/parse.h source/sema.h
+)
+
+add_library(cpp2util INTERFACE)
+add_library(cppfront::cpp2util ALIAS cpp2util)
+target_compile_features(cpp2util INTERFACE cxx_std_20)
+target_include_directories(
+    cpp2util INTERFACE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+)
+target_sources(
+    cpp2util
+    INTERFACE
+    FILE_SET HEADERS
+    BASE_DIRS include
+    FILES include/cpp2util.h
+)
+
+if (NOT PROJECT_IS_TOP_LEVEL)
+    get_directory_property(_CPPFRONT_MAGIC_DIR PARENT_DIRECTORY)
+    include(cmake/CppfrontHelpers.cmake)
+endif ()
+
+if (CPPFRONT_INSTALL_RULES)
+    include(GNUInstallDirs)
+    include(CMakePackageConfigHelpers)
+
+    set(CPPFRONT_INSTALL_CMAKEDIR "${CMAKE_INSTALL_DATADIR}/cmake/cppfront"
+        CACHE STRING "Install location for cppfront's platform-independent CMake files")
+
+    set(CPPFRONT_INSTALL_EXE_CMAKEDIR "${CMAKE_INSTALL_LIBDIR}/cmake/cppfront-exe"
+        CACHE STRING "Install location for cppfront's platform-specific CMake files")
```

Comment:
I'd mark these advanced personally. The defaults are what one would expect, so fiddling with them is not really expected.

Reply:
Sure